domain/machine/state/state_test.go

```diff
@@ -496,3 +496,52 @@ func (s *stateSuite) TestIsControllerNotFound(c *gc.C) {
 	_, err := s.state.IsController(context.Background(), "666")
 	c.Assert(err, jc.ErrorIs, machineerrors.NotFound)
 }
+
+// TestMarkMachineForRemovalSuccess asserts the happy path of
+// MarkMachineForRemoval at the state layer.
+func (s *stateSuite) TestMarkMachineForRemovalSuccess(c *gc.C) {
+	err := s.state.CreateMachine(context.Background(), "666", "", "123")
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = s.state.MarkMachineForRemoval(context.Background(), "666")
+	c.Check(err, jc.ErrorIsNil)
+
+	var mark bool
+	err = s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		return tx.QueryRowContext(ctx, "SELECT mark_for_removal FROM machine_removals WHERE machine_uuid=?", "123").Scan(&mark)
+	})
+	c.Check(err, jc.ErrorIsNil)
+	c.Assert(mark, gc.Equals, true)
+}
+
+// TestMarkMachineForRemovalNotFound asserts that a NotFound error is returned
+// when the machine is not found.
+// TODO(cderici): use machineerrors.MachineNotFound on rebase after #17759
+// lands.
+func (s *stateSuite) TestMarkMachineForRemovalNotFound(c *gc.C) {
+	err := s.state.MarkMachineForRemoval(context.Background(), "666")
+	c.Assert(err, jc.ErrorIs, machineerrors.NotFound)
+}
+
+// TestAllMachineRemovalsSuccess asserts the happy path of AllMachineRemovals at
+// the state layer.
+func (s *stateSuite) TestAllMachineRemovalsSuccess(c *gc.C) {
+	err := s.state.CreateMachine(context.Background(), "666", "", "123")
```

Comment:
**suggestion:** To avoid dumb regression, it may be useful to also have a test with several machine selected to removal, with some machine not selected for removal in the machine set.